src/main/java/bio/terra/service/filedata/azure/util/BlobCrl.java

```diff
@@ -109,8 +114,35 @@ public BlobContainerCopier createBlobContainerCopier(URL sourceUrl, String desti
    *
    * @param blobName blob name to delete.
    */
-  public void deleteBlob(String blobName) {
-    blobContainerClientFactory.getBlobContainerClient().getBlobClient(blobName).delete();
+  public boolean deleteBlob(String blobName) {
+    try {
+      blobContainerClientFactory.getBlobContainerClient().getBlobClient(blobName).delete();
+      return true;
+    } catch (BlobStorageException e) {
+      if (e.getStatusCode() == HttpStatus.NOT_FOUND.value()) {
```

Comment:
Would it be better to check if the blob exists instead of waiting for an exception to be thrown? Using exceptions for control flow is a bit of an anti-pattern.

Reply:
Even if you add a check, you still need to catch, because someone could delete it after you check. Therefore it seems better to keep the code as-is. This is typical situation when dealing with any shared resource without locking.

Reply:
Ah good point. Silly me with my single-threaded brain!